docs/pages/build-reference/variables.mdx

```diff
@@ -167,143 +194,27 @@ After creating a secret, you can read it on subsequent EAS Build jobs with `proc
-In **app.config.js**, we may be using the API URL like this:
-
-```js app.config.js
-export default {
-  // ...
-  extra: {
-    // Fall back to development URL when not set
-    apiUrl: process.env.API_URL ?? 'https://localhost:3000'
-    enableHiddenFeatures: process.env.ENABLE_HIDDEN_FEATURES ? Boolean(process.env.ENABLE_HIDDEN_FEATURES) : true,
-  }
-}
-```
-
-Using this approach, we would always need to remember to run `API_URL=https://api.staging.com ENABLE_HIDDEN_FEATURES=1 eas update` when updating staging, and something similar for production. If we forgot the `ENABLE_HIDDEN_FEATURES=0` flag when publishing to production, we might end up rolling out untested features to production, and if we forgot the `API_URL` value, then users would be pointed to `https://localhost:3000`!
-
-The following are two possible alternative approaches, each with different tradeoffs.
-
-<Step label="1">
-
-**Move values to application code and switch based on channel**. Rather than putting configuration in environment variables and extras, create a JavaScript file, possibly named **Config.js**. This approach will work well for you as long as you don't need to use the configuration values to modify build time configuration, such as the `ios.bundleIdentifier`, `icon`, and so on. This approach also gives you the ability to promote updates between environments, because the configuration that is used will switch when it's loaded from a binary with a different channel. It might look something like this:
-
-<Collapsible summary="Config.js">
-
-```js
-import * as Updates from 'expo-updates';
-
-let Config = {
-  apiUrl: 'https://localhost:3000',
-  enableHiddenFeatures: true,
-};
-
-if (Updates.channel === 'production') {
-  Config.apiUrl = 'https://api.production.com';
-  Config.enableHiddenFeatures = false;
-} else if (Updates.channel === 'staging') {
-  Config.apiUrl = 'https://api.staging.com';
-  Config.enableHiddenFeatures = true;
-}
-
-export default Config;
-```
-
-</Collapsible>
-
-</Step>
-
-<Step label="2">
-
-**Use a single environment variable to toggle configuration**. In our **eas.json** we can set an environment variable such as `APP_ENV` and then switch on that value inside of **app.config.js**. This way, we only have to be sure to set one environment variable: `APP_ENV=production eas update`.
-
-<Collapsible summary="eas.json">
-
-```json
-{
-  "build": {
-    "production": {
-      "channel": "production",
-      "env": {
-        "APP_ENV": "production"
-      }
-    },
-    "preview": {
-      "channel": "staging",
-      "env": {
-        "APP_ENV": "staging"
-      }
-    }
-  }
-}
-```
-
-</Collapsible>
+### Can EAS Build use .env files?
+Environment variables defined in a **.env** file are only considered by the Expo CLI. Therefore, if you upload a **.env** file to EAS Build, it will be used to inline `EXPO_PUBLIC_` variables into your application code, but will not be using for evaluating your **app.config.js** or for any other build steps.
```

Comment:
Good call! Cleaned this up and added some advice based on Brent and I just talking about this to clarify that the most predictable way to avoid differences here is to use .env files everywhere but EAS Build.